### DISCUSSION TITLE
Arrow Array Stream import/export?

### DISCUSSION THREAD

**Discussion by eitsupi** [apache/datafusion#10598] [Q&A] [answered]
I wanted to create a MemTable from an Array Stream, but realized that MemTable does not expose the internal Arrow data and I can only access the original data by executing a query.
https://github.com/apache/datafusion/blob/f80794779ef251528ee983e22ec03c158deb42a6/datafusion-examples/examples/memtable.rs#L30-L52

Which Struct that DataFusion has would be ideal if I want to input/output data?
(I mean I was able to input data into MemTable and was hoping to be able to output the same, but apparently not, so I'm stumped)

**Comment by jonahgao** [comment]
Do you want to access the data inside the `Memtable` directly without using a query?
I'm not entirely sure, but I didn't find such a structure; it seems you might need to consider [customizing](https://github.com/apache/datafusion/blob/f80794779ef251528ee983e22ec03c158deb42a6/datafusion-examples/examples/custom_datasource.rs) a `TableProvider` based on `Memtable`.

**Reply by eitsupi** [reply, marked as answer]
Thanks for your reply.
Finally, I judged it reasonable to define a simple Struct with C Stream interface.

```rust
struct Table {
    pub schema: std::sync::Arc<arrow::datatypes::Schema>,
    pub batches: Vec<datafusion::arrow::array::RecordBatch>,
}
```
